File: backend/src/main/java/pt/isel/rugby/model/Practice.java
package pt.isel.rugby.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Table(name = "practice")
@Data
public class Practice implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private Date date;

    @Column
    private String local;

    @Column
    private String comment;

    @OneToMany(mappedBy = "practice")
    private List<AthletePractice> athletePractices;
}

File: frontend2/Rugby/src/app/classes/practice.ts
import {AthletePractice} from './associations/AthletePractice';

export class Practice {
  constructor(
      public id?: number,
      public date?: Date,
      public local?: string,
      public comment?: string,
      public athletePractices?: AthletePractice[]
  ) {
    this.id = id ? id : 0;
    this.comment = comment ? comment : '';
    this.date = date ? date : new Date(0);
    this.local = local ? local : '';
    this.athletePractices = athletePractices ? athletePractices : [];
  }
}
